From 04e16df5d3ae1b5f5c76dd4bd4bfcbc424e85b58 Mon Sep 17 00:00:00 2001
From: xzl <xiangzelong@deepin.org>
Date: Tue, 19 Sep 2023 16:18:56 +0800
Subject: [PATCH 2/2] Add huawei patch to support pangu machine

---
 tools/hciattach.c | 43 +++++++++++++++++++++++++++++++++++++++++++
 tools/hciattach.h |  1 +
 2 files changed, 44 insertions(+)

diff --git a/tools/hciattach.c b/tools/hciattach.c
index 276a4e5..78eb530 100644
--- a/tools/hciattach.c
+++ b/tools/hciattach.c
@@ -1097,7 +1097,13 @@ struct uart_t uart[] = {
 	{ "amp",	0x0000, 0x0000, HCI_UART_H4, 115200, 115200,
 			AMP_DEV, DISABLE_PM, NULL, NULL, NULL },
 
+	/*Hisi Bluetooth Module */
+	{ "hisi",		0x0000, 0x0000, HCI_UART_H4,   115200, 4000000,
+			FLOW_CTL, DISABLE_PM, NULL, NULL, NULL },
+
 	{ NULL, 0 }
+
+
 };
 
 static struct uart_t * get_by_id(int m_id, int p_id)
@@ -1210,6 +1216,31 @@ fail:
 	return -1;
 }
 
+/* Initialize UART driver */
+static int hisi_init(char *dev, struct uart_t *u)
+{
+	printf("hisi_init\n");
+	int fd;
+	fd = open(dev, O_RDWR);
+	printf("dev open, fd = %d\n",fd);
+
+	if (fd < 0) {
+		perror("Can't open device hisi");
+		goto fail;
+	}
+
+	if (ioctl(fd, BT_IOCTL_HCISETPROTO, u->proto) < 0) {
+		perror("Can't set device hisi");
+		goto fail;
+	}
+
+	return fd;
+
+fail:
+	close(fd);
+	return -1;
+}
+
 static void usage(void)
 {
 	printf("hciattach - HCI UART driver initialization utility\n");
@@ -1346,6 +1377,18 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
+	/*hisi init*/
+	if(u->type == "hisi"){
+		n = hisi_init(dev,u);
+
+		if (n < 0) {
+			perror("Can't initialize device");
+			exit(1);
+		}
+
+		return 0;
+	}
+
 	/* If user specified a initial speed, use that instead of
 	   the hardware's default */
 	if (init_speed)
diff --git a/tools/hciattach.h b/tools/hciattach.h
index dfa4c1e..c23af90 100644
--- a/tools/hciattach.h
+++ b/tools/hciattach.h
@@ -14,6 +14,7 @@
 #define N_HCI	15
 #endif
 
+#define BT_IOCTL_HCISETPROTO     101
 #define HCIUARTSETPROTO		_IOW('U', 200, int)
 #define HCIUARTGETPROTO		_IOR('U', 201, int)
 #define HCIUARTGETDEVICE	_IOR('U', 202, int)
-- 
2.33.1
