Description: fix main.conf
Author: wangyouwan <wangyouwan@uniontech.com>
Origin: https://gerrit.uniontech.com/plugins/gitiles/base/bluez/+/d174ea3ff37a84009c7dcce89cca09834266d2f2
Last-Update: 2022-05-20

--- bluez-5.64.orig/src/main.conf
+++ bluez-5.64/src/main.conf
@@ -12,7 +12,7 @@
 # How long to stay in discoverable mode before going back to non-discoverable
 # The value is in seconds. Default is 180, i.e. 3 minutes.
 # 0 = disable timer, i.e. stay discoverable forever
-#DiscoverableTimeout = 0
+DiscoverableTimeout = 0
 
 # Always allow pairing even if there are no agent registered
 # Possible values: true, false
@@ -68,7 +68,7 @@
 # however the tradeoff is increased power consumptions. This feature
 # will fully work only on kernel version 4.1 and newer. Defaults to
 # 'false'.
-#FastConnectable = false
+FastConnectable = true
 
 # Default privacy setting.
 # Enables use of private address.
